ci/gitlab-ci/publish-docker/flavor-1/.gitlab-ci.yml

```
@@ -0,0 +1,15 @@
+stages:
+  - publish
+publish:
+  stage: publish
+  image: docker:19.03.1
+  services:
+    - docker:19.03.1-dind
+  variables:
+    DOCKER_TLS_CERTDIR: "/certs"
+  before_script:
+    - apk add --repository https://alpine.secrethub.io/alpine/edge/main --allow-untrusted secrethub-cli
+  script:
+    - secrethub read company/app/docker/pasword | docker login -u $(secrethub read company/app/docker/username) --password-stdin
+    - docker build -t company/app:${CI_COMMIT_SHA:0:7} .
```

And I'd say let's swap these: first do `docker build` then `docker login` + `push`)

```suggestion
    - docker build -t company/app:${CI_COMMIT_SHA:0:7} .
    - secrethub read company/app/docker/password | docker login -u $(secrethub read company/app/docker/username) --password-stdin
```